cvat/apps/consensus/views.py

```diff
@@ -0,0 +1,468 @@
+            ),
+        },
+    )
+    def create(self, request, *args, **kwargs):
+        self.check_permissions(request)
+        input_serializer = ConsensusReportCreateSerializer(data=request.data)
+        input_serializer.is_valid(raise_exception=True)
+        task_id = input_serializer.validated_data["task_id"]
+
+        queue_name = settings.CVAT_QUEUES.CONSENSUS.value
+        queue = django_rq.get_queue(queue_name)
+        rq_id = request.query_params.get(self.CREATE_REPORT_RQ_ID_PARAMETER, None)
+
+        if rq_id is None:
+            try:
+                task = Task.objects.get(pk=task_id)
+            except Task.DoesNotExist as ex:
+                raise NotFound(f"Task {task_id} does not exist") from ex
+
+            try:
+                rq_id = scehdule_consensus_merging(task, request)
+                serializer = RqIdSerializer({"rq_id": rq_id})
+                return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+            except Exception as ex:
+                raise ValidationError(str(ex))
+
+        else:
+            rq_job = queue.fetch_job(rq_id)
+            if (
+                not rq_job
+                or not ConsensusReportPermission.create_scope_check_status(
+                    request, job_owner_id=rq_job.meta["user"]["id"]
+                )
+                .check_access()
+                .allow
+            ):
+                # We should not provide job existence information to unauthorized users
+                raise NotFound("Unknown request id")
+
+            if rq_job.is_failed:
+                message = str(rq_job.exc_info)
+                rq_job.delete()
+                raise ValidationError(message)
+            elif rq_job.is_queued or rq_job.is_started:
+                return Response(status=status.HTTP_202_ACCEPTED)
+            elif rq_job.is_finished:
+                return_value = rq_job.return_value()
+                rq_job.delete()
+                if not return_value:
+                    raise ValidationError("No report has been computed")
+
+                report = self.get_queryset().get(pk=return_value)
+                report_serializer = ConsensusReportSerializer(instance=report)
+                return Response(
+                    data=report_serializer.data,
+                    status=status.HTTP_201_CREATED,
+                    headers=self.get_success_headers(report_serializer.data),
+                )
+
+    @extend_schema(
+        operation_id="consensus_retrieve_report_data",
+        summary="Get consensus report contents",
+        responses={"200": OpenApiTypes.OBJECT},
+    )
+    @action(detail=True, methods=["GET"], url_path="data", serializer_class=None)
+    def data(self, request, pk):
+        report = self.get_object()  # check permissions
+        json_report = prepare_report_for_downloading(report, host=get_server_url(request))
+        return HttpResponse(json_report.encode())
```

Comment:
```suggestion
        return HttpResponse(json_report.encode(), content_type="application/json")
```
By default, it returns an HTML document.